File: dotnet/RelevantCodes.ExtentReports/RelevantCodes.ExtentReports/View/Icon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RelevantCodes.ExtentReports.View
{
    public class Icon
    {
        public static string GetIcon(LogStatus Status)
        {
            switch (Enum.GetName(typeof(LogStatus), Status).ToLower())
            {
                case "fail": return "mdi-navigation-cancel";
                case "fatal": return "mdi-navigation-cancel";
                case "error": return "mdi-alert-error";
                case "warning": return "mdi-alert-warning";
                case "pass": return "mdi-action-check-circle";
                case "info": return "mdi-action-info-outline";
                case "skip": return "mdi-av-skip-next";
                default: return "mdi-action-help";
            }
        }
    }
}

File: ruby/lib/extent/icon.rb
require 'rubygems'
require 'liquid'

module RelevantCodes
	class Icon < Liquid::Drop
		def self.icon(status)
			case status
			when 'pass' 
				ret_val = 'mdi-action-check-circle'
			when 'fail' 
				ret_val = 'mdi-navigation-cancel'
			when 'fatal' 
				ret_val = 'mdi-navigation-cancel'
			when 'error' 
				ret_val = 'mdi-alert-error'
			when 'warning' 
				ret_val = 'mdi-alert-warning'
			when 'skip' 
				ret_val = 'mdi-content-redo'
			when 'info' 
				ret_val = 'mdi-action-info-outline'
			else 
				ret_val = 'mdi-action-help'
			end

			ret_val
		end
	end
end
